packages/core/src/external/aws/lambda-logic/bundle-to-brief.ts

```diff
@@ -341,67 +483,114 @@ function applyResourceSpecificFilters(res: Resource): any | undefined {
         return undefined;
       }
 
-      updRes.vaccineCode = getUniqueDisplays(res.vaccineCode);
+      updRes.vaccineCode = getUniqueDisplaysString(res.vaccineCode);
     }
+    if (res.site?.text) {
+      updRes.site = res.site.text;
+    }
+    updRes.route = getUniqueDisplaysString(res.route);
 
-    delete res.doseQuantity;
+    delete updRes.doseQuantity;
   }
 
   if (res.resourceType === "Practitioner") {
     const name = getNameString(res.name);
     updRes.name = name;
 
-    const qualificationsText = getUniqueDisplays(res.qualification?.[0]?.code);
+    const qualificationsText = getLongestDisplay(res.qualification?.[0]?.code);
     if (qualificationsText) updRes.qualification = qualificationsText;
 
     updRes.address = getAddressString(res.address);
   }
 
   if (res.resourceType === "Procedure") {
-    const name = getUniqueDisplays(res.code);
+    const name = getUniqueDisplaysString(res.code);
     if (name) {
       updRes.name = name;
       if (name.includes("no data")) return undefined;
     }
 
     delete updRes.code;
 
-    if (updRes.status === "") delete updRes.status;
+    if (isUselessStatus(updRes.status)) delete updRes.status;
 
-    delete res.reasonCode; // TODO: Introduce term server lookup here
+    delete updRes.reasonCode; // TODO: #2510 - Introduce term server lookup here
+    delete updRes.report;
+    delete updRes.note;
   }
 
   if (res.resourceType === "DiagnosticReport") {
-    const code = res.code?.text;
-    if (code) updRes.code = code;
+    const mostDescriptiveType = getLongestDisplay(res.code);
+    if (mostDescriptiveType) updRes.type = mostDescriptiveType;
+    if (isUselessStatus(updRes.status)) delete updRes.status;
+
+    const allTypes = getUniqueDisplays(res.code);
+    if (allTypes) {
+      let removeResource = false;
+      allTypes.forEach(type => {
+        if (
+          REPORT_TYPES_BLACKLIST.includes(type) ||
+          REPORT_TYPES_BLACKLIST.some(blacklistedType => type.includes(blacklistedType))
+        )
+          removeResource = true;
+      });
+      if (removeResource) return undefined;
```

Comment:
nit: could just use a regular loop and return if the condition is met, no need for variable and additional `if`

Reply:
yup, 💯